metadata/metadata.go

```diff
@@ -1727,6 +1827,270 @@ func (serv *MetadataServer) GetEntities(stream pb.Metadata_GetEntitiesServer) er
 	})
 }
 
+func (serv *MetadataServer) CreateTrigger(ctx context.Context, trigger *pb.Trigger) (*pb.Empty, error) {
+	return serv.genericCreate(ctx, &triggerResource{trigger}, nil)
+}
+
+func (serv *MetadataServer) AddTrigger(ctx context.Context, tr *pb.TriggerRequest) (*pb.Empty, error) {
```

Comment:
Lets walk through the add and remove when you get the other stuff cleaned up. Theres probably cleaner ways to do these functions

Reply:
+10 on this, you can delegate the creation to each variant instead of putting all the logic in here